sdk/core/System.ClientModel/tests/client/TestClients/SimpleClient/SimpleClientServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Net.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace ClientModel.ReferenceClients.SimpleClient;
+
+public static class SimpleClientServiceCollectionExtensions
+{
+    // TODO: How much of these can SCM provide for SCM-based clients to use
+    // from a central source?  Microsoft.Extensions.ClientModel, e.g.?
+
+    // No parameters uses client defaults
+    // See: https://learn.microsoft.com/en-us/dotnet/core/extensions/options-library-authors#parameterless
+    public static IServiceCollection AddSimpleClient(this IServiceCollection services)
+    {
+        services.AddLogging();
+
+        // Add client options
+        services.AddOptions<SimpleClientOptions>()
+            .Configure<ILoggerFactory>((options, loggerFactory)
+                => options.Logging.LoggerFactory = loggerFactory);
+
+        // Proxy to logging options in case a custom logging policy is added
+        services.AddSingleton<ClientLoggingOptions>(sp =>
+        {
+            IOptions<SimpleClientOptions> options = sp.GetRequiredService<IOptions<SimpleClientOptions>>();
+            return options.Value.Logging;
+        });
+
+        services.AddSingleton<SimpleClient>(sp =>
+        {
+            IConfiguration configuration = sp.GetRequiredService<IConfiguration>();
+            IConfiguration clientConfiguration = configuration.GetSection("SimpleClient");
+
+            Uri endpoint = clientConfiguration.GetValue<Uri>("ServiceUri") ??
+                throw new InvalidOperationException("Expected 'ServiceUri' to be present in 'SimpleClient' configuration settings.");
+
+            // TODO: how to get this securely?
+            ApiKeyCredential credential = new("fake key");
+
+            // TODO: to roll a credential, this will need to be IOptionsMonitor
+            // not IOptions -- come back to this.
```

Comment:
Do we even roll credentials via IConfiguration?  What does the Azure KeyVault client story look like here to refresh secrets at runtime?